### PR TITLE
Add a new option --scrub-all-chroots

### LINKED ISSUE #521: Add command to remove all existing chroots
### Short description of the problem
I'm using mock for plenty of things, to test packages, to test builds, test my code behavior on a different system versions. Problem is that when you use mock heavily it's easy to get lost in which chroot still exists and your free space will leak this way.

I know you can look on /var/lib/mock and remove that one by one. However, this is not really inconvenient  when you have hundreds of chroots.

Nice solution have podman and buildah implemented already by ``--all`` option:

https://www.mankier.com/1/podman-rm#--all

### Output of `rpm -q mock`

mock-2.0-2.fc31.noarch

### Steps to reproduce issue

1. mock -r fedora-rawhide-x86_64 --init
2. <future> mock --remove-all-chroots

### PR DESCRIPTION
**Pull request by praiskup**
Fixes #521

### CODE REVIEW

**Comment by xsuchy** [comment]
I would rather have it inside of mock.py and utilize the logic we have from `--list-chroots`.

**Comment by praiskup** [comment]
I'm not against modifying `mock.py` per se, but `--list-chroots` doesn't know the previously used `--uniqueext` extensions.  This was rather a toy experiment for now.

**Comment by praiskup** [comment]
@xsuchy does this sound better? :-)

File: mock/py/mockbuild/scrub_all.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+from mockbuild.constants import MOCKCONFDIR
+from mockbuild.config import traverse_chroot_configs
+
+
+def _do_scrub(configs, weird, chroot, suffix=None):
+    if suffix and chroot + "-" + suffix in weird:
+        print(f"skipping weird scrub: {chroot} {suffix}")
+        return
+
+    base_cmd = ["mock", "--scrub=all", "-r"]
```
Comment (xsuchy) [inline mock/py/mockbuild/scrub_all.py:19]: ??? Now you can call the function directly. Or is there an obstacle?

**Comment by xsuchy** [comment]
Otherwise mucho bettero.

File: mock/py/mockbuild/scrub_all.py (hunk 1 above)
Comment (praiskup) [reply mock/py/mockbuild/scrub_all.py:19]: That would mean calling the following multiple times:

```python
commands = Commands(config_opts, uidManager, plugins, state, buildroot, bootstrap_buildroot)
commands.scrub(all) 
```
Which is _a lot_ of work (manpower, to code properly all the arguments)  :-/ my opinion is :shrug: keep it as is this is mostly cosmetics.  Admitedly that's not nice, but still better than nothing.

**Comment by praiskup** [comment]
This deserves some documentation work, if acceptable at all.  @xsuchy lemme know what you think so I could finalize this PR.

**Comment by praiskup** [comment]
Please take a look.

File: mock/py/mock.py [hunk 2]
```diff
@@ -87,6 +87,7 @@
 from mockbuild.state import State
 from mockbuild.trace_decorator import traceLog
 import mockbuild.uid
+from mockbuild.scrub_all import scrub_all_chroots
```
[the remark this answers is not in the document]
Comment (praiskup) [reply mock/py/mock.py:80]: Ok, fixing this was easy, actually (separate commit).

File: mock/py/mockbuild/scrub_all.py (hunk 1 above)
Comment (praiskup) [reply mock/py/mockbuild/scrub_all.py:20]: Added the FIXME note at least.

**Review by xsuchy — APPROVED** [review]
+1

**Pull request merged**